docs/rest_api.rst

```diff
@@ -0,0 +1,175 @@
+
+
+Course Access Groups
+--------------------
+
+These endpoints allow to create, edit and delete Course Access Group.
+
+List Groups
+~~~~~~~~~~~
+This endpoint returns a paginated list of json objects in "results".
+Each object representing a single Course Access Group.
+
+.. note::
+
+    In this version there's no filtering mechanism.
+
+.. code-block:: bash
+
+    GET /course_access_groups/api/v1/course-access-groups/
+
+    {
+      "count": 50,
+      "next": "http://mydomain.com/course_access_groups/api/v1/course-access-groups/?limit=20&offset=20",
+      "previous": null,
+      "results": [
+        {
+          "id": 1,
+          "name": "Customers",
+          "description": "Any customer should be enrolled here"
+        },
+        {
+          "id": 2,
+          "name": "Sales Employees",
+          "description": "All team members from the sales team"
+        }
+      ]
+    }
+
+Group Details
+~~~~~~~~~~~~~
+This endpoint provides a detailed view of a single Course Access Group when
+providing an group identifier (``id`` for short).
+
+
+.. note::
+
+    At In this version this endpoint offers no additional information compared to
+    the list view.
+
+
+.. code-block:: bash
+
+    GET /course_access_groups/api/v1/course-access-groups/2/
+
+    {
+      "id": 2,
+      "name": "Sales Employees",
+      "description": "All team members from the sales team"
+    }
+
+Add a New Group
+~~~~~~~~~~~~~~~
+Performing a ``POST`` request to this endpoint allows to add a new group.
+Both of the ``name`` and the ``description`` ``POST`` parameters are required.
+
+.. code-block:: bash
+
+    POST /course_access_groups/api/v1/course-access-groups/
+    {"name": "New Group", "description": "My new group"}
+
+
+Modify a Group
+~~~~~~~~~~~~~~
+To modify a group, a ``PATCH`` request can be used.
+Both ``name`` and ``description`` can be changed.
+
+.. note::
+
+    This endpoint requires a ``Content-Type: application/json`` and the
+    request payload to be a properly formatted json object as shown below.
+
+
+.. code-block:: bash
+
+    PATCH /course_access_groups/api/v1/course-access-groups/2/
+    {"name": "Awesome Group"}
+
+
+Delete a Group
+~~~~~~~~~~~~~~
+``DELETE`` request can be used for deletion, albeit one group at a time.
+
+.. note::
+
+    In this version, ``DELETE`` is not idempotent, in which deleting an object
+    twice will result in a 404 code for the next request. This is not really
+    a problem as much as it of an issue of not conforming to the HTTP
+    standards.
+
+
+.. code-block:: bash
+
+    DELETE /course_access_groups/api/v1/course-access-groups/2/
+
+
+Linking Courses to Course Access Groups
+---------------------------------------
+
+These endpoints allow to add and remove courses from Course Access Groups.
+
+
+.. note::
+
+    This section is still work in progress.
+
+
+Marking Courses as Public
+-------------------------
+
+These endpoints allow making courses as a public course that is available to
```

Comment:
"These endpoints allow us to make" or "These endpoints mark courses as public. This means the course is available to all site learners regardless of their..."